validations/src/validations.rs

```diff
@@ -1112,37 +1189,89 @@ pub fn validate_tally_transaction<'a>(
                 .into());
             }
         } else {
-            if reveals_count > 0 {
-                // Make sure every rewarded address is a revealer
-                if dr_state.info.reveals.get(&output.pkh).is_none() {
-                    return Err(TransactionError::RevealNotFound.into());
+            if _after_second_hard_fork {
+                if honests_count > 0 {
+                    // TODO: non-revealers can now be rewarded, but only if the tally result is
+                    // InsufficientConsensus
+
+                    // Make sure every rewarded address is a revealer
+                    if dr_state.info.reveals.get(&output.pkh).is_none() {
+                        return Err(TransactionError::RevealNotFound.into());
+                    }
+                    // Make sure every rewarded address passed the tally function, a.k.a. "is honest" / "is not a liar"
```

Comment:
should we include this check conditions in a function? Something like check_revealer(dr_state, output) -> Result((), TransactionError).

On this way, the code could be more readable, because we only have to put:
check_revealer(..)?;

Reply:
I tried to extract all the logic from inside the for loop into a "validate_tally_output" function, but it needs 10 arguments so I'm not sure if it's worth it.

Reply:
my proposal was one function by each conditional branch, but I am also not sur if it's worth it